executor/src/native_executor.rs

```diff
@@ -15,20 +15,26 @@
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
 use error::{Error, ErrorKind, Result};
-use native_runtime as runtime;
 use runtime_io;
 use state_machine::{Externalities, CodeExecutor};
 use wasm_executor::WasmExecutor;
 
 use std::panic::catch_unwind;
 
-pub struct NativeExecutor;
+pub trait NativeExecutionDispatch {
+	fn native_equivalent() -> &'static [u8];
+	fn dispatch(method: &str, data: &[u8]) -> Option<Vec<u8>>;
+}
+
+pub struct NativeExecutor<D: NativeExecutionDispatch + Sync + Send> {
+	pub _dummy: ::std::marker::PhantomData<D>,
```

Comment:
docs missing. also the public `_dummy` field seems strange

Reply:
needed to introduce the `D` type somewhere - just doing `pub struct NativeExecutor<D: NativeExecutionDispatch + Sync + Send>;` got me a compile error.

Reply:
Yes, what I mean is that for now the `NativeExecutor` type existing at all is unecessary